apps/meteor/server/services/moderation/service.ts

```diff
@@ -0,0 +1,48 @@
+import { ServiceClassInternal } from '@rocket.chat/core-services';
+import type { IModerationService } from '@rocket.chat/core-services';
+import { Permissions, Roles, Users } from '@rocket.chat/models';
+
+import { createOrUpdateProtectedRoleAsync } from '../../lib/roles/createOrUpdateProtectedRole';
+
+export class ModerationService extends ServiceClassInternal implements IModerationService {
+	protected name = 'moderation';
+
+	async resetUserRoles(roles: string[]): Promise<void> {
+		const usersInRole = await Users.findUsersInRoles(roles, undefined, { projection: { _id: 1 } }).toArray();
+
+		const usersToRemove = usersInRole.map((user) => user._id);
```

Comment:
you can call map on the cursor directly

Reply:
Sorry, I did not understand this comment; what would this do?